File: BE/models/Review.js
const mongoose = require('mongoose');

const reviewSchema = new mongoose.Schema({
  reviewId: {
    type: String,
    unique: true,
    required: true
  },
  userId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'User',
    required: true
  },
  serviceId: {
    type: mongoose.Schema.Types.ObjectId,
    ref: 'Service',
    required: true
  },
  rating: {
    type: Number,
    required: true,
    min: 1,
    max: 5
  },
  comment: {
    type: String,
    required: true
  },
  photos: [{
    type: String
  }],
  isVerified: {
    type: Boolean,
    default: false
  }
}, {
  timestamps: true
});

// Index
reviewSchema.index({ serviceId: 1, createdAt: -1 });
reviewSchema.index({ userId: 1 });

module.exports = mongoose.model('Review', reviewSchema);

File: fe/lib/models/review.dart
class Review {
  final String id;
  final String reviewId;
  final String userId;
  final String serviceId;
  final double rating;
  final String comment;
  final List<String>? photos;
  final bool isVerified;
  final DateTime createdAt;

  // User info (populated)
  final String? userName;
  final String? userAvatar;

  Review({
    required this.id,
    required this.reviewId,
    required this.userId,
    required this.serviceId,
    required this.rating,
    required this.comment,
    this.photos,
    this.isVerified = false,
    required this.createdAt,
    this.userName,
    this.userAvatar,
  });

  factory Review.fromJson(Map<String, dynamic> json) {
    return Review(
      id: json['_id'] ?? '',
      reviewId: json['reviewId'] ?? '',
      userId: json['userId'] ?? '',
      serviceId: json['serviceId'] ?? '',
      rating: (json['rating'] ?? 0).toDouble(),
      comment: json['comment'] ?? '',
      photos: json['photos'] != null ? List<String>.from(json['photos']) : null,
      isVerified: json['isVerified'] ?? false,
      createdAt: DateTime.parse(
        json['createdAt'] ?? DateTime.now().toIso8601String(),
      ),
      userName: json['user']?['name'],
      userAvatar: json['user']?['avatarUrl'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      '_id': id,
      'reviewId': reviewId,
      'userId': userId,
      'serviceId': serviceId,
      'rating': rating,
      'comment': comment,
      'photos': photos,
      'isVerified': isVerified,
      'createdAt': createdAt.toIso8601String(),
    };
  }
}
